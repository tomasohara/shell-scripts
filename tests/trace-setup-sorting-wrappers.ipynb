{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "65d43538",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PROMPT_DKQQARMRPJAU\\[\\]>\n"
     ]
    }
   ],
   "source": [
    "## TEST:  Make sure simple prompt used (e.g., no escapes that might contaminate output)\n",
    "## PS1=\"$ \"\n",
    "## TODO: PS1=\"> \"\n",
    "\n",
    "## NOTE: The Jupyter bash kernel requires that PS1 not be modified as they customize it. \n",
    "echo $PS1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d623c9a9",
   "metadata": {},
   "outputs": [],
   "source": [
    "## BAD\n",
    "# # Delete all aliases and function\n",
    "# # TODO: Instead start from pristine environment\n",
    "# unalias -a\n",
    "# alias | wc -l\n",
    "# for f in $(typeset -f | egrep '^\\w+'); do unset -f $f; done\n",
    "# typeset -f | egrep '^\\w+' | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "0ff8f157",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Setting a temp directory for tests\n",
    "# TMP=/tmp/test-setupsort\n",
    "\n",
    "## NOTE: Source it directly from the ./tests directory.\n",
    "## OLD: BIN_DIR=$PWD/..\n",
    "\n",
    "## OLD\n",
    "# alias | wc -l | test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "54d25281",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Global Setup\n",
    "\n",
    "# alias testnum=\"sed -r \"s/[0-9]/H/g\"\" \n",
    "alias testuser=\"sed -r \"s/\"$USER\"+/userxf333/g\"\"\n",
    "alias testnum=\"sed -r \"s/[0-9][0-9]*/N/g\"\"\n",
    "## For number of digits not being critical"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ab954c7f",
   "metadata": {},
   "outputs": [],
   "source": [
    "## NOTE: For reproducability, the directory name needs to be fixed\n",
    "## In place of $$, use a psuedo random number (e,g., 6411)\n",
    "## *** All output from one run to the next needs to be the same ***\n",
    "\n",
    "## temp_dir=$TMP/test-$$\n",
    "TMP=${TMP:-/tmp}\n",
    "if [ \"$DEBUG_LEVEL\" -lt 4 ]; then TMP=/tmp/test-setup-sorting-wrappers; fi\n",
    "temp_dir=$TMP/trace-setup-sorting-wrappers\n",
    "rename-with-file-date \"$temp_dir\"\n",
    "\n",
    "## OLD\n",
    "# mkdir -p \"$temp_dir\"\n",
    "# # TODO: /bin/rm -rvf \"$temp_dir\"\n",
    "# cd \"$temp_dir\"\n",
    "# pwd\n",
    "\n",
    "## ALIAS FOR PRINTING SEPERATION LINES (FOR JUPYTER)\n",
    "# alias linebr=\"printf '%*s\\n' \"${COLUMNS:-$(tput cols)}\" '' | tr ' ' -\"\n",
    "\n",
    "command mkdir -p \"$temp_dir\"\n",
    "# TODO: /bin/rm -rvf \"$temp_dir\"\n",
    "command cd \"$temp_dir\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ec6df159",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N\n"
     ]
    }
   ],
   "source": [
    "## Count aliases proper\n",
    "## OLD\n",
    "# alias | wc -l\n",
    "alias | wc -l | testnum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "23c3b474",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N\n"
     ]
    }
   ],
   "source": [
    "## Count functions\n",
    "## OLD\n",
    "typeset -f | egrep '^\\w+' | wc -l | testnum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "badd6828",
   "metadata": {},
   "outputs": [],
   "source": [
    "## BAD\n",
    "# source $BIN_DIR/tomohara-aliases.bash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "78c13815",
   "metadata": {},
   "outputs": [],
   "source": [
    "## OLD\n",
    "# pwd | testuser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c91d1116",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# FOR SOME REASONS \"cat bashrc-setup\" COMMAND GIVES AN ERROR\n",
    "\n",
    "# cat ~/.bashrc | tab-sort\n",
    "# linebr\n",
    "# cat ~/.bashrc | colon-sort\n",
    "# linebr\n",
    "# cat ~/.bashrc | colon-sort-rev-num\n",
    "# linebr\n",
    "# cat ~/.bashrc | freq-sort\n",
    "# linebr\n",
    "\n",
    "## OLD: rm -rf ./* > /dev/null\n",
    "\n",
    "## OLD\n",
    "# man printf > pf_manual.txt\n",
    "\n",
    "## OLD\n",
    "# cat pf_manual.txt | tab-sort | testnum | testuser\n",
    "# man uname > uname_manual.txt\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "a405288e",
   "metadata": {},
   "outputs": [],
   "source": [
    "man uname > uname_manual.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "f440c180",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True\n"
     ]
    }
   ],
   "source": [
    "cat uname_manual.txt | grep -q \"uname\" && echo \"True\" || echo \"False\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "19316ba3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False\n"
     ]
    }
   ],
   "source": [
    "## FalseCase: To show that grep-echo pipeline works as usual\n",
    "man printf | grep -q \"uname\" && echo \"True\" || echo \"False\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "776f4c45",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True\n"
     ]
    }
   ],
   "source": [
    "## TrueCase: To show that grep-echo pipeline works as usual\n",
    "man printf | grep -q \"printf\" && echo \"True\" || echo \"False\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "312d103e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True\n"
     ]
    }
   ],
   "source": [
    "## NEW: Added tail pipeline\n",
    "# cat uname_manual.txt | colon-sort | testnum | testuser | tail -n 1\n",
    "cat uname_manual.txt | colon-sort | grep -q \"uname\" && echo \"True\" || echo \"False\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "28e58950",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True\n"
     ]
    }
   ],
   "source": [
    "## NEW: Added tail pipeline\n",
    "cat uname_manual.txt | colon-sort-rev-num | grep -q \"uname\" && echo \"True\" || echo \"False\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "f113e7d6-eebf-4080-8472-0aee9a72fd13",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5:50\n",
      "10:20\n"
     ]
    }
   ],
   "source": [
    "# Data for sorting\n",
    "echo $'10:20\\n5:50' > _data_.list\n",
    "colon-sort -n _data_.list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "333b2760-2134-48e8-8621-4d05bb019661",
   "metadata": {},
   "outputs": [],
   "source": [
    "# The first column is sorted by default"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "b15d7d7c-79c1-44e5-8b84-751ae0d1a107",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n"
     ]
    }
   ],
   "source": [
    "colon-sort -n _data_.list | head -1 | grep -c 5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "a8f3c7d0-bd44-430a-89e1-0847ee844c69",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10:20\n"
     ]
    }
   ],
   "source": [
    "colon-sort --key=2 -n _data_.list | head -1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "5d4b26be-2c2a-484c-b2a1-8e558f894683",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n"
     ]
    }
   ],
   "source": [
    "colon-sort --key=2 -n _data_.list | head -1 | grep -v -c 5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "cd321795-8cd4-4342-8e58-095193b28d16",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False\n"
     ]
    }
   ],
   "source": [
    "## NEW: Added tail pipeline\n",
    "## TODO2: rework to use data for which freq-sort and para-sort makes sense\n",
    "## OLD: cat uname_manual.txt | freq-sort | grep -q \"uname\" && echo \"True\" || echo \"False\"\n",
    "cat uname_manual.txt | freq-sort | head -3 | grep -q \"uname\" && echo \"True\" || echo \"False\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "bcaf053a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ECHOES THE FILE \n",
    "uname -o > linux-info.txt\n",
    "\n",
    "## OLD: Seperated commands increases test score\n",
    "# cat linux-info.txt | echoize\n",
    "# $ cat ~/.bashrc | echoize"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "e3539e1d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GNU/Linux\n"
     ]
    }
   ],
   "source": [
    "# TODO2: use multi-line input to echoize\n",
    "cat linux-info.txt | echoize"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "4270d960",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False\n"
     ]
    }
   ],
   "source": [
    "# SORTS BY PARAGRAPH\n",
    "## OLD: cat uname_manual.txt | para-sort | grep -q \"uname\" && echo \"True\" || echo \"False\"\n",
    "cat uname_manual.txt | para-sort | head | grep -q \"uname\" && echo \"True\" || echo \"False\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "6e12887f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True\n"
     ]
    }
   ],
   "source": [
    "cat uname_manual.txt | echoize | grep -q \"uname\" && echo \"True\" || echo \"False\""
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
