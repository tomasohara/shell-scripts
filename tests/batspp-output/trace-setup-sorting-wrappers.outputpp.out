1..15
not ok 1 test-1
# (in test file batspp-output/trace-setup-sorting-wrappers.outputpp, line 25)
#   `[ "$(test-1-actual)" == "$(test-1-expected)" ]' failed
# ========== echo $PS1
#  ==========
#
# ========= [PEXP\[\]ECT_PROMPT> =========
# [PEXP\[\]ECT_PROMPT>
# ============================
not ok 2 test-3
# (in test file batspp-output/trace-setup-sorting-wrappers.outputpp, line 59)
#   `[ "$(test-3-actual)" == "$(test-3-expected)" ]' failed
# ========== unalias -a
#  ==========
# ========= $ alias | wc -l\n$ for f in ; do unset -f ; done\n$ typeset -f | egrep '^\w+' | wc -l\n0\n0 =========
# $ alias | wc -l
# $ for f in bats_abort_timeout_countdown ()
# bats_add_debug_exclude_path ()
# bats_all_in ()
# bats_any_in ()
# bats_append_arrays_as_args ()
# bats_approx_realpath ()
# bats_binary_search ()
# bats_capture_stack_trace ()
# bats_check_status_from_trap ()
# bats_check_valid_version ()
# bats_create_test_tmpdirs ()
# bats_debug_trap ()
# bats_emit_trace ()
# bats_error_trap ()
# bats_evaluate_preprocessed_source ()
# bats_execute ()
# bats_exit_trap ()
# bats_extract_line ()
# bats_format_file_line_reference ()
# bats_format_file_line_reference_colon ()
# bats_format_file_line_reference_comma_line ()
# bats_format_file_line_reference_uri ()
# bats_frame_filename ()
# bats_frame_function ()
# bats_frame_lineno ()
# bats_generate_warning ()
# bats_get_child_processes_of ()
# bats_get_failure_stack_trace ()
# bats_internal_load ()
# bats_interrupt_trap ()
# bats_interrupt_trap_in_run ()
# bats_kill_childprocesses_of ()
# bats_load_library ()
# bats_load_library_safe ()
# bats_load_safe ()
# bats_merge_stdout_and_stderr ()
# bats_normalize_windows_dir_path ()
# bats_perform_test ()
# bats_prefix_lines_for_tap_output ()
# bats_preprocess_source ()
# bats_print_failed_command ()
# bats_print_stack_trace ()
# bats_quote_code ()
# bats_readlinkf ()
# declare -fx bats_readlinkf
# bats_redirect_stderr_into_file ()
# bats_replace_filename ()
# bats_require_minimum_version ()
# bats_separate_lines ()
# bats_setup_tracing ()
# bats_should_retry_test ()
# bats_sort ()
# bats_start_timeout_countdown ()
# bats_strip_string ()
# bats_teardown_trap ()
# bats_test_begin ()
# bats_test_function ()
# bats_timeout_trap ()
# bats_trim ()
# bats_trim_filename ()
# bats_version_lt ()
# bats_warn_minimum_guaranteed_version ()
# find_in_bats_lib_path ()
# get_mills_since_epoch ()
# load ()
# run ()
# setup ()
# skip ()
# teardown ()
# test-1-actual ()
# test-1-expected ()
# END_EXPECTED
# test-10-actual ()
# test-10-expected ()
# END_EXPECTED
# test-11-actual ()
# test-11-expected ()
# END_EXPECTED
# test-12-actual ()
# test-12-expected ()
# SYNOPSIS
# SEE ALSO
# REPORTING BUGS
# PRINTF(H)                        User Commands                       PRINTF(H)
# NAME
# GNU coreutils H.HH               February HHHH                       PRINTF(H)
# DESCRIPTION
# COPYRIGHT
# AUTHOR
# END_EXPECTED
# test-13-actual ()
# test-13-expected ()
# SYNOPSIS
# SEE ALSO
# REPORTING BUGS
# PRINTF(H)                        User Commands                       PRINTF(H)
# NAME
# GNU coreutils H.HH               February HHHH                       PRINTF(H)
# DESCRIPTION
# COPYRIGHT
# AUTHOR
# END_EXPECTED
# test-14-actual ()
# test-14-expected ()
# H.HH.H-HH-generic
# END_EXPECTED
# test-15-actual ()
# test-15-expected ()
# END_EXPECTED
# test-16-actual ()
# test-16-expected ()
# PRINTF(H)                        User Commands                       PRINTF(H)
# END_EXPECTED
# test-3-actual ()
# test-3-expected ()
# 0
# 0
# END_EXPECTED
# test-4-actual ()
# test-4-expected ()
# 0
# END_EXPECTED
# test-5-actual ()
# test-5-expected ()
# END_EXPECTED
# test-6-actual ()
# test-6-expected ()
# 1
# 0
# END_EXPECTED
# test-7-actual ()
# test-7-expected ()
# END_EXPECTED
# test-8-actual ()
# test-8-expected ()
# END_EXPECTED
# test-9-actual ()
# test-9-expected ()
# END_EXPECTED
# test_test-2d1 ()
# test_test-2d10 ()
# test_test-2d11 ()
# test_test-2d12 ()
# SYNOPSIS
# SEE ALSO
# REPORTING BUGS
# PRINTF(H)                        User Commands                       PRINTF(H)
# NAME
# GNU coreutils H.HH               February HHHH                       PRINTF(H)
# DESCRIPTION
# COPYRIGHT
# AUTHOR
# test_test-2d13 ()
# SYNOPSIS
# SEE ALSO
# REPORTING BUGS
# PRINTF(H)                        User Commands                       PRINTF(H)
# NAME
# GNU coreutils H.HH               February HHHH                       PRINTF(H)
# DESCRIPTION
# COPYRIGHT
# AUTHOR
# test_test-2d14 ()
# H.HH.H-HH-generic' "=========";
# test_test-2d15 ()
# test_test-2d16 ()
# test_test-2d3 ()
# test_test-2d4 ()
# 0' "=========";
# test_test-2d5 ()
# test_test-2d6 ()
# test_test-2d7 ()
# test_test-2d8 ()
# test_test-2d9 () ; do unset -f ; done
# $ typeset -f | egrep '^\w+' | wc -l
# 0
# 0
# ============================
not ok 3 test-4
# (in test file batspp-output/trace-setup-sorting-wrappers.outputpp, line 97)
#   `[ "$(test-4-actual)" == "$(test-4-expected)" ]' failed
# ========== TMP=/tmp/test-setupsort
#  ==========
# ========= ## NOTE: Source it directly from the ./tests directory.
# $ BIN_DIR=$PWD/..
# $ alias | wc -l
# 0 =========
# ## NOTE: Source it directly from the ./tests directory.
# $ BIN_DIR=/tmp/batspp-76755/test-4/..
# $ alias | wc -l
# 0
# ============================
not ok 4 test-5
# (in test file batspp-output/trace-setup-sorting-wrappers.outputpp, line 134)
#   `[ "$(test-5-actual)" == "$(test-5-expected)" ]' failed
# ========== temp_dir=$TMP/test-6411
#  ==========
# ========= $ mkdir -p "$temp_dir"
# # TODO: /bin/rm -rvf "$temp_dir"
# $ cd "$temp_dir"
# $ pwd
# #ALIAS FOR PRINTING SEPERATION LINES (FOR JUPYTER)
# $ alias linebr="printf '%*s\n' "${COLUMNS:-$(tput cols)}" '' | tr ' ' -"
# /tmp/test-setupsort/test-6411 =========
# $ mkdir -p "/test-6411"
# # TODO: /bin/rm -rvf "/test-6411"
# $ cd "/test-6411"
# $ pwd
# #ALIAS FOR PRINTING SEPERATION LINES (FOR JUPYTER)
# $ alias linebr="printf '%*s\n' "80" '' | tr ' ' -"
# /tmp/test-setupsort/test-6411
# ============================
not ok 5 test-6
# (in test file batspp-output/trace-setup-sorting-wrappers.outputpp, line 174)
#   `[ "$(test-6-actual)" == "$(test-6-expected)" ]' failed
# ========== alias | wc -l
#  ==========
# 0
# ========= # Count functions\n$ typeset -f | egrep '^\w+' | wc -l\n1\n0 =========
# # Count functions
# $ typeset -f | egrep '^\w+' | wc -l
# 1
# 0
# ============================
not ok 6 test-7
# (in test file batspp-output/trace-setup-sorting-wrappers.outputpp, line 211)
#   `[ "$(test-7-actual)" == "$(test-7-expected)" ]' failed
# ========== alias testnum="sed -r "s/[0-9]/H/g""
#  ==========
# ========= $ alias testuser="sed -r "s/"$USER"+/userxf333/g"" =========
# $ alias testuser="sed -r "s/"aveey"+/userxf333/g""
# ============================
not ok 7 test-8
# (from function `test-8-actual' in file batspp-output/trace-setup-sorting-wrappers.outputpp, line 252,
#  in test file batspp-output/trace-setup-sorting-wrappers.outputpp, line 239)
#   `test-8-actual ' failed
# ========== source $BIN_DIR/tomohara-aliases.bash
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-setup-sorting-wrappers.outputpp: line 252: /tomohara-aliases.bash: No such file or directory
not ok 8 test-9
# (from function `test-9-actual' in file batspp-output/trace-setup-sorting-wrappers.outputpp, line 286,
#  in test file batspp-output/trace-setup-sorting-wrappers.outputpp, line 273)
#   `test-9-actual ' failed with status 127
# ========== pwd | testuser
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-setup-sorting-wrappers.outputpp: line 286: testuser: command not found
not ok 9 test-10
# (in test file batspp-output/trace-setup-sorting-wrappers.outputpp, line 313)
#   `[ "$(test-10-actual)" == "$(test-10-expected)" ]' failed
# ========== rm -rf ./* > /dev/null
#  ==========
# ========= $ man printf > pf_manual.txt
# $ cat pf_manual.txt | tab-sort | testnum | testuser =========
# $ man printf > pf_manual.txt
# $ cat pf_manual.txt | tab-sort | testnum | testuser
# ============================
not ok 10 test-11
# (from function `test-11-actual' in file batspp-output/trace-setup-sorting-wrappers.outputpp, line 355,
#  in test file batspp-output/trace-setup-sorting-wrappers.outputpp, line 342)
#   `test-11-actual ' failed with status 127
# ========== cat pf_manual.txt | colon-sort | testnum | testuser
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-setup-sorting-wrappers.outputpp: line 355: colon-sort: command not found
# cat: pf_manual.txt: No such file or directory
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-setup-sorting-wrappers.outputpp: line 355: testnum: command not found
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-setup-sorting-wrappers.outputpp: line 355: testuser: command not found
not ok 11 test-12
# (from function `test-12-actual' in file batspp-output/trace-setup-sorting-wrappers.outputpp, line 389,
#  in test file batspp-output/trace-setup-sorting-wrappers.outputpp, line 376)
#   `test-12-actual ' failed with status 127
# ========== cat pf_manual.txt | colon-sort-rev-num | testnum | testuser
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-setup-sorting-wrappers.outputpp: line 389: colon-sort-rev-num: command not found
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-setup-sorting-wrappers.outputpp: line 389: testnum: command not found
# cat: pf_manual.txt: No such file or directory
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-setup-sorting-wrappers.outputpp: line 389: testuser: command not found
not ok 12 test-13
# (from function `test-13-actual' in file batspp-output/trace-setup-sorting-wrappers.outputpp, line 476,
#  in test file batspp-output/trace-setup-sorting-wrappers.outputpp, line 463)
#   `test-13-actual ' failed with status 127
# ========== cat pf_manual.txt | freq-sort | testnum | testuser
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-setup-sorting-wrappers.outputpp: line 476: freq-sort: command not found
# cat: pf_manual.txt/home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-setup-sorting-wrappers.outputpp: line 476: testnum: command not found
# : No such file or directory
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-setup-sorting-wrappers.outputpp: line 476: testuser: command not found
not ok 13 test-14
# (in test file batspp-output/trace-setup-sorting-wrappers.outputpp, line 556)
#   `[ "$(test-14-actual)" == "$(test-14-expected)" ]' failed
# ========== uname -r > linux-info.txt
#  ==========
# ========= $ cat linux-info.txt | echoize | testnum | testuser
# # $ cat ~/.bashrc | echoize
# H.HH.H-HH-generic =========
# $ cat linux-info.txt | echoize | testnum | testuser
# # $ cat ~/.bashrc | echoize
# H.HH.H-HH-generic
# ============================
not ok 14 test-15
# (from function `test-15-actual' in file batspp-output/trace-setup-sorting-wrappers.outputpp, line 599,
#  in test file batspp-output/trace-setup-sorting-wrappers.outputpp, line 586)
#   `test-15-actual ' failed with status 127
# ========== cat pf_manual.txt | para-sort | testnum | testuser
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-setup-sorting-wrappers.outputpp: line 599: para-sort: command not found
# cat: pf_manual.txt/home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-setup-sorting-wrappers.outputpp: line 599: testnum: command not found
# : No such file or directory
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-setup-sorting-wrappers.outputpp: line 599: testuser: command not found
not ok 15 test-16
# (from function `test-16-actual' in file batspp-output/trace-setup-sorting-wrappers.outputpp, line 633,
#  in test file batspp-output/trace-setup-sorting-wrappers.outputpp, line 620)
#   `test-16-actual ' failed with status 127
# ========== cat pf_manual.txt | echoize | head -n 5 | testuser | testnum
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-setup-sorting-wrappers.outputpp: line 633: echoize: command not found
# cat: pf_manual.txt: No such file or directory
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-setup-sorting-wrappers.outputpp: line 633: testuser: command not found
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-setup-sorting-wrappers.outputpp: line 633: testnum: command not found
